### PR TITLE
Implementation of bidirection kill for kill filter.

### PR DESCRIPTION
**Pull request by KBaichoo**
Signed-off-by: Kevin Baichoo <kbaichoo@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/main/PULL_REQUESTS.md)

Commit Message: Modify the kill filter to support directionality.
Additional Description: This allows us to more easily test response crashes.
Risk Level: low (this is a test-only filter AFAIK)
Testing: unit test, integration test
Docs Changes: Included
Release Notes: Included
Platform Specific Features: N/A
Related Issue #14249 

/assign @qqustc 

Since you're one of the maintainers of the filter. PTAL.

### CODE REVIEW

**Review by qqustc — APPROVED** [review]

**Comment by KBaichoo** [comment]
@alyssawilk since you're the maintainer on-call, can you assign maintainer to review these changes? @qqustc is an extension owner and has already signed off. Thanks!

File: source/extensions/filters/http/kill_request/kill_request_filter.cc
Comment (htuch) [inline source/extensions/filters/http/kill_request/kill_request_filter.cc:49]: Nit: `const bool`

File: source/extensions/filters/http/kill_request/kill_request_filter.cc
Comment (htuch) [inline source/extensions/filters/http/kill_request/kill_request_filter.cc:82]: Ditto.

File: api/envoy/extensions/filters/http/kill_request/v3/kill_request.proto
Comment (htuch) [inline api/envoy/extensions/filters/http/kill_request/v3/kill_request.proto:26]: Question on whether to call this REQUEST/RESPONSE or DECODE/ENCODE?

**Review by htuch — CHANGES_REQUESTED** [review]
LGTM modulo minor comments.
/wait

File: api/envoy/extensions/filters/http/kill_request/v3/kill_request.proto
Comment (antoniovicente) [reply api/envoy/extensions/filters/http/kill_request/v3/kill_request.proto:26]: decode and encode happen in each direction.  Vote for REQUEST/RESPONSE

Comment (KBaichoo) [reply api/envoy/extensions/filters/http/kill_request/v3/kill_request.proto]: Went with REQUEST/RESPONSE

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
